### ISSUE TITLE
[Intellij]: Can not debug and run my simple flutter app on Chrome

### ISSUE THREAD

**Issue by KyloEntro** [flutter/flutter#99543]
Hi,

I have Flutter 2.10.3 and I use a Mac m1 with Jetbrains IntelliJ, last released version. 
I can not run and debug my app when the target is Web - Chrome (last version available).

## Steps to Reproduce
- Create a new project in Jetbrains IntelliJ
- Run the app by setting the target "web"

Chrome opened and display a blank page and in the console I get :
```
Failed to set DevTools server address: ext.flutter.activeDevToolsServerAddress: (-32603) ext.flutter.activeDevToolsServerAddress: Unexpected error from chrome devtools:
text: Uncaught (in promise)
exception:
  description: null
  type: string
  value: Assertion failed: "boolean expression must not be null"
attempted JS eval: `require("dart_sdk").developer.invokeExtension(
    "ext.flutter.activeDevToolsServerAddress", JSON.stringify({"value":"http://127.0.0.1:9100"}));
`

#0      handleErrorIfPresent (package:dwds/src/utilities/shared.dart:97:5)
#1      ChromeProxyService.callServiceExtension (package:dwds/src/services/chrome_proxy_service.dart:351:5)
<asynchronous suspension>
#2      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>
. Deep links to DevTools will not show in Flutter errors.
ext.flutter.connectedVmServiceUri: (-32603) ext.flutter.connectedVmServiceUri: Unexpected error from chrome devtools:
text: Uncaught (in promise)
exception:
  description: null
  type: string
  value: Assertion failed: "boolean expression must not be null"
attempted JS eval: `require("dart_sdk").developer.invokeExtension(
    "ext.flutter.connectedVmServiceUri", JSON.stringify({"value":"http://127.0.0.1:62279/AtF_6AE-OHs="}));
`

#0      handleErrorIfPresent (package:dwds/src/utilities/shared.dart:97:5)
#1      ChromeProxyService.callServiceExtension (package:dwds/src/services/chrome_proxy_service.dart:351:5)
<asynchronous suspension>
#2      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>

Failed to set vm service URI: ext.flutter.connectedVmServiceUri: (-32603) ext.flutter.connectedVmServiceUri: Unexpected error from chrome devtools:
text: Uncaught (in promise)
exception:
  description: null
  type: string
  value: Assertion failed: "boolean expression must not be null"
attempted JS eval: `require("dart_sdk").developer.invokeExtension(
    "ext.flutter.connectedVmServiceUri", JSON.stringify({"value":"http://127.0.0.1:62279/AtF_6AE-OHs="}));
`

#0      handleErrorIfPresent (package:dwds/src/utilities/shared.dart:97:5)
#1      ChromeProxyService.callServiceExtension (package:dwds/src/services/chrome_proxy_service.dart:351:5)
<asynchronous suspension>
#2      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>
. Deep links to DevTools will not show in Flutter errors.
```

I didn't have this issue when I tried on iOS and Android.

**Comment by darshankawar** [comment]
@KyloEntro 
Thanks for the report. Although I don't have M1, but I tried this scenario on Intel mac using latest stable (2.10.3) by running a sample app in web which compiled and built properly without any issues:

```
dhs@Dhss-MacBook-Pro testdemo % flutter run -d chrome
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             17.6s
This app is linked to the debug service: ws://127.0.0.1:52919/qG80BAUaoi0=/ws
Debug service listening on ws://127.0.0.1:52919/qG80BAUaoi0=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:52919/qG80BAUaoi0=
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:52919/qG80BAUaoi0=
```

My Intellij IDEA config:

<img width="468" alt="Screenshot 2022-03-04 at 6 46 09 PM" src="https://user-images.githubusercontent.com/67046386/156770240-0ec55766-d090-48a8-a607-54a1e093c69f.png">


Can you provide `flutter doctor -v` ?

**Comment by KyloEntro** [comment]
Here flutter doctor -v output :  
```
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.2.1 21D62 darwin-arm, locale
    fr-FR)
    • Flutter version 2.10.3 at /Users/leoji/Tools/flutter_2.10
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (il y a 2 jours), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
    • Android SDK at /Users/leoji/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.5744.122

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.2.1 21D62 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 99.0.4844.51

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

My IntelliJ config : 
<img width="485" alt="Capture d’écran 2022-03-04 à 14 21 45" src="https://user-images.githubusercontent.com/64692412/156770956-d4999930-a2fa-45f8-84f2-5507e69c102c.png">

and : 
```
flutter run -d chrome 
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             14,6s
This app is linked to the debug service: ws://127.0.0.1:56256/OTa0-DcCBrM=/ws
Debug service listening on ws://127.0.0.1:56256/OTa0-DcCBrM=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at:
http://127.0.0.1:56256/OTa0-DcCBrM=
Assertion failed: "boolean expression must not be null"
    at Object.wrapException
    (http://localhost:56202/dwds/src/injected/client.js:1095:17)
    at Object.assertThrow
    (http://localhost:56202/dwds/src/injected/client.js:1539:15)
    at Object.boolConversionCheck
    (http://localhost:56202/dwds/src/injected/client.js:1535:11)
    at main__closure1.call$2
    (http://localhost:56202/dwds/src/injected/client.js:24571:13)
    at Object.Primitives_applyFunction
    (http://localhost:56202/dwds/src/injected/client.js:967:30)
    at Object.Function_apply
    (http://localhost:56202/dwds/src/injected/client.js:4954:16)
    at _callDartFunctionFast
    (http://localhost:56202/dwds/src/injected/client.js:7515:16)
    at http://localhost:56202/dwds/src/injected/client.js:7506:18
    at Object._postEvent (http://localhost:56202/dart_sdk.js:52107:19)
    at Object.postEvent (http://localhost:56202/dart_sdk.js:52095:15)
    at binding$5.WidgetsFlutterBinding.new.binding.BindingBase.new
    (http://localhost:56202/packages/flutter/src/foundation/binding.dart.lib.js:
    320:15)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_GestureBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53249:49)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_SchedulerBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53255:51)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_ServicesBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53261:50)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_PaintingBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53267:50)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_SemanticsBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53273:51)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_RendererBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53279:50)
    at binding$5.WidgetsFlutterBinding.new.BindingBase_WidgetsBinding$36.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53285:49)
    at new binding$5.WidgetsFlutterBinding.new
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53298:51)
    at Function.ensureInitialized
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53290:54)
    at Object.runApp
    (http://localhost:56202/packages/flutter/src/widgets/widget_inspector.dart.l
    ib.js:53307:44)
    at main$
    (http://localhost:56202/packages/test_chrome/main.dart.lib.js:339:13)
    at main (http://localhost:56202/web_entrypoint.dart.lib.js:33:29)
    at main.next (<anonymous>)
    at http://localhost:56202/dart_sdk.js:40571:33
    at _RootZone.runUnary (http://localhost:56202/dart_sdk.js:40441:59)
    at _FutureListener.thenAwait.handleValue
    (http://localhost:56202/dart_sdk.js:35363:29)
    at handleValueCallback (http://localhost:56202/dart_sdk.js:35931:49)
    at Function._propagateToListeners
    (http://localhost:56202/dart_sdk.js:35969:17)
    at _Future.new.[_completeWithValue]
    (http://localhost:56202/dart_sdk.js:35817:23)
    at http://localhost:56202/dart_sdk.js:34998:46
    at _RootZone.runUnary (http://localhost:56202/dart_sdk.js:40441:59)
    at _FutureListener.then.handleValue
    (http://localhost:56202/dart_sdk.js:35363:29)
    at handleValueCallback (http://localhost:56202/dart_sdk.js:35931:49)
    at Function._propagateToListeners
    (http://localhost:56202/dart_sdk.js:35969:17)
    at _Future.new.[_completeWithValue]
    (http://localhost:56202/dart_sdk.js:35817:23)
    at async._AsyncCallbackEntry.new.callback
    (http://localhost:56202/dart_sdk.js:35838:35)
    at Object._microtaskLoop (http://localhost:56202/dart_sdk.js:40708:13)
    at _startMicrotaskLoop (http://localhost:56202/dart_sdk.js:40714:13)
    at http://localhost:56202/dart_sdk.js:36191:9
══╡ EXCEPTION CAUGHT BY FLUTTER FRAMEWORK
╞═════════════════════════════════════════════════════════
Failed to set DevTools server address: ext.flutter.activeDevToolsServerAddress:
(-32000) Server error
{"exception":"Assertion failed: \"boolean expression must not be
null\"","stack":"dwds/src/injected/client.js 1095:17
wrapException\ndwds/src/injected/client.js 1539:15
assertThrow\ndwds/src/injected/client.js 1535:11
boolConversionCheck\ndwds/src/injected/client.js 24571:13
call$2\ndwds/src/injected/client.js 967:30
Primitives_applyFunction\ndwds/src/injected/client.js 4954:16
Function_apply\ndwds/src/injected/client.js 7515:16
_callDartFunctionFast\ndwds/src/injected/client.js 7506:18
<fn>\ndart-sdk/lib/_internal/js_dev_runtime/patch/developer_patch.dart 134:45
_postEvent\ndart-sdk/lib/developer/extension.dart 151:3
postEvent\npackages/flutter/src/foundation/binding.dart 483:15
postEvent\npackages/flutter/src/foundation/binding.dart 469:5
[_postExtensionStateChangedEvent]\npackages/flutter/src/foundation/binding.dart
512:11
<fn>\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50
<fn>\ndart-sdk/lib/async/zone.dart 1685:54
runUnary\ndart-sdk/lib/async/future_impl.dart 159:18
handleValue\ndart-sdk/lib/async/future_impl.dart 766:44
handleValueCallback\ndart-sdk/lib/async/future_impl.dart 795:13
_propagateToListeners\ndart-sdk/lib/async/future_impl.dart 601:5
[_completeWithValue]\ndart-sdk/lib/async/future_impl.dart 639:7
callback\ndart-sdk/lib/async/schedule_microtask.dart 40:11
_microtaskLoop\ndart-sdk/lib/async/schedule_microtask.dart 49:5
_startMicrotaskLoop\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dar
t 166:15      <fn>\n","method":"ext.flutter.activeDevToolsServerAddress"}. Deep
links to DevTools will not show in Flutter errors.
The following LegacyJavaScriptObject object was thrown during a service
extension callback for
"ext.flutter.activeDevToolsServerAddress":
  Assertion failed: "boolean expression must not be null"

When the exception was thrown, this was the stack:
dwds/src/injected/client.js 1095:17
wrapException
dwds/src/injected/client.js 1539:15
assertThrow
dwds/src/injected/client.js 1535:11
boolConversionCheck
dwds/src/injected/client.js 24571:13                                     call$2
dwds/src/injected/client.js 967:30
Primitives_applyFunction
dwds/src/injected/client.js 4954:16
Function_apply
dwds/src/injected/client.js 7515:16
_callDartFunctionFast
dwds/src/injected/client.js 7506:18                                      <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/developer_patch.dart 134:45
_postEvent
dart-sdk/lib/developer/extension.dart 151:3
postEvent
packages/flutter/src/foundation/binding.dart 483:15
postEvent
packages/flutter/src/foundation/binding.dart 469:5
[_postExtensionStateChangedEvent]
packages/flutter/src/foundation/binding.dart 512:11                      <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50       <fn>
dart-sdk/lib/async/zone.dart 1685:54
runUnary
dart-sdk/lib/async/future_impl.dart 159:18
handleValue
dart-sdk/lib/async/future_impl.dart 766:44
handleValueCallback
dart-sdk/lib/async/future_impl.dart 795:13
_propagateToListeners
dart-sdk/lib/async/future_impl.dart 601:5
[_completeWithValue]
dart-sdk/lib/async/future_impl.dart 639:7
callback
dart-sdk/lib/async/schedule_microtask.dart 40:11
_microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5
_startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15      <fn>
════════════════════════════════════════════════════════════════════════════════
════════════════════
Another exception was thrown: Assertion failed: "boolean expression must not be
null"
ext.flutter.connectedVmServiceUri: (-32000) Server error
{"exception":"Assertion failed: \"boolean expression must not be
null\"","stack":"dwds/src/injected/client.js 1095:17
wrapException\ndwds/src/injected/client.js 1539:15
assertThrow\ndwds/src/injected/client.js 1535:11
boolConversionCheck\ndwds/src/injected/client.js 24571:13
call$2\ndwds/src/injected/client.js 967:30
Primitives_applyFunction\ndwds/src/injected/client.js 4954:16
Function_apply\ndwds/src/injected/client.js 7515:16
_callDartFunctionFast\ndwds/src/injected/client.js 7506:18
<fn>\ndart-sdk/lib/_internal/js_dev_runtime/patch/developer_patch.dart 134:45
_postEvent\ndart-sdk/lib/developer/extension.dart 151:3
postEvent\npackages/flutter/src/foundation/binding.dart 483:15
postEvent\npackages/flutter/src/foundation/binding.dart 469:5
[_postExtensionStateChangedEvent]\npackages/flutter/src/foundation/binding.dart
512:11
<fn>\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50
<fn>\ndart-sdk/lib/async/zone.dart 1685:54
runUnary\ndart-sdk/lib/async/future_impl.dart 159:18
handleValue\ndart-sdk/lib/async/future_impl.dart 766:44
handleValueCallback\ndart-sdk/lib/async/future_impl.dart 795:13
_propagateToListeners\ndart-sdk/lib/async/future_impl.dart 601:5
[_completeWithValue]\ndart-sdk/lib/async/future_impl.dart 639:7
callback\ndart-sdk/lib/async/schedule_microtask.dart 40:11
_microtaskLoop\ndart-sdk/lib/async/schedule_microtask.dart 49:5
_startMicrotaskLoop\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dar
t 166:15      <fn>\n","method":"ext.flutter.connectedVmServiceUri"}
Failed to set vm service URI: ext.flutter.connectedVmServiceUri: (-32000) Server
error
{"exception":"Assertion failed: \"boolean expression must not be
null\"","stack":"dwds/src/injected/client.js 1095:17
wrapException\ndwds/src/injected/client.js 1539:15
assertThrow\ndwds/src/injected/client.js 1535:11
boolConversionCheck\ndwds/src/injected/client.js 24571:13
call$2\ndwds/src/injected/client.js 967:30
Primitives_applyFunction\ndwds/src/injected/client.js 4954:16
Function_apply\ndwds/src/injected/client.js 7515:16
_callDartFunctionFast\ndwds/src/injected/client.js 7506:18
<fn>\ndart-sdk/lib/_internal/js_dev_runtime/patch/developer_patch.dart 134:45
_postEvent\ndart-sdk/lib/developer/extension.dart 151:3
postEvent\npackages/flutter/src/foundation/binding.dart 483:15
postEvent\npackages/flutter/src/foundation/binding.dart 469:5
[_postExtensionStateChangedEvent]\npackages/flutter/src/foundation/binding.dart
512:11
<fn>\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50
<fn>\ndart-sdk/lib/async/zone.dart 1685:54
runUnary\ndart-sdk/lib/async/future_impl.dart 159:18
handleValue\ndart-sdk/lib/async/future_impl.dart 766:44
handleValueCallback\ndart-sdk/lib/async/future_impl.dart 795:13
_propagateToListeners\ndart-sdk/lib/async/future_impl.dart 601:5
[_completeWithValue]\ndart-sdk/lib/async/future_impl.dart 639:7
callback\ndart-sdk/lib/async/schedule_microtask.dart 40:11
_microtaskLoop\ndart-sdk/lib/async/schedule_microtask.dart 49:5
_startMicrotaskLoop\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dar
t 166:15      <fn>\n","method":"ext.flutter.connectedVmServiceUri"}. Deep links
to DevTools will not show in Flutter errors.
The Flutter DevTools debugger and profiler on Chrome is available at:
http://127.0.0.1:9100?uri=http://127.0.0.1:56256/OTa0-DcCBrM=
```

**Comment by darshankawar** [comment]
Thanks for the details. Since I don't have M1, and unable to replicate on intel, I'll label it for further insights from the team.

/cc @yjbanov @christopherfujino

**Comment by KyloEntro** [comment]
To my mind, it is not linked with IntelliJ because I have the same issue when I tried to run it on my terminal with flutter run.

**Comment by KyloEntro** [comment]
I didn't have this issue with flutter 2.10.0

**Comment by tkreuder** [comment]
I also have the issue with: 2.10.3 and also with 2.10.0

```
Failed to set DevTools server address: ext.flutter.activeDevToolsServerAddress: (-32603) ext.flutter.activeDevToolsServerAddress: WipError -32000 Inspected target navigated or closed
. Deep links to DevTools will not show in Flutter errors.
ext.flutter.connectedVmServiceUri: (-32603) ext.flutter.connectedVmServiceUri: Bad state: No context with the running Dart application.
#0      RemoteDebuggerExecutionContext.id (package:dwds/src/debugging/execution_context.dart:55:7)
<asynchronous suspension>
#1      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>

Failed to set vm service URI: ext.flutter.connectedVmServiceUri: (-32603) ext.flutter.connectedVmServiceUri: Bad state: No context with the running Dart application.
#0      RemoteDebuggerExecutionContext.id (package:dwds/src/debugging/execution_context.dart:55:7)
<asynchronous suspension>
#1      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>
. Deep links to DevTools will not show in Flutter errors.
```

flutter master or dev channel did not solve it either. As the stack trace is not super helpful, it is hard to find out the cause of it.

**Comment by elliette** [comment]
Hi @KyloEntro! I just tried to reproduce this on a Mac with an M1 chip and Flutter `v.2.10.3` and wasn't able to. I noticed you are on an older version of Chrome than I am (I'm on `99.0.4844.84`). Do you mind updating your Chrome browser and check if this still reproduces?

Another thing to try: there is a new Flutter stable available (version `2.10.4`). Could you try updating your Flutter version as well?

Thank you!

**Comment by KyloEntro** [comment]
I updated my flutter to 2.10.4, all plugins in my IntelliJ and chrome all work ! 

Also, I no longer have the package "archive" issue, before 2.10.4 I have to change the version of "archive" in flutter installation to use it in IntelliJ. 

Thanks !
